File: com/bubblesort/rosetta/rosetta.java
package com.bubblesort.rosetta;

import java.util.Random;

public class rosetta{
    public static void bubbleSort(int[] comparable) { // public static <E extends Comparable<? super E>> void bubbleSort(E[] comparable)
        boolean changed = false;
        do {
            changed = false;
            for (int a = 0; a < comparable.length - 1; a++) {
                if (comparable[a] > comparable[a + 1]) { // (comparable[a].compareTo(comparable[a + 1]) > 0)
                    int tmp = comparable[a];                 //E tmp = comparable[a];
                    comparable[a] = comparable[a + 1];
                    comparable[a + 1] = tmp;
                    changed = true;
                }
            }
        } while (changed);
    }

    public static void main(String[] args) {
        // Criando um array com 1000 valores
        int[] array = new int[10000];

        // Preenchendo o array com valores aleatórios entre 1 e 100
        Random random = new Random();
        for (int i = 0; i < array.length; i++) {
            array[i] = random.nextInt(100) + 1; // Gera um número aleatório entre 1 e 100
        }

        // Chamando o método de ordenação
        bubbleSort(array);
    }
}

File: C/bubblesort/rosetta/rosetta.c
#include <stdio.h>

void bubble_sort (int *a, int n) {
    int i, t, j = n, s = 1;
    while (s) {
        s = 0;
        for (i = 1; i < j; i++) {
            if (a[i] < a[i - 1]) {
                t = a[i];
                a[i] = a[i - 1];
                a[i - 1] = t;
                s = 1;
            }
        }
        j--;
    }
}

int main() {
    int n = 1000;
    int a[n];
    srand(time(NULL));

    for (int i = 0; i < n; i++) {
        a[i] = rand() % 10000;  // Gera valores aleatórios entre 0 e 9999
    }

    bubble_sort(a, n);

    return 0;
}